tao_compiler/mlir/disc/transforms/lower_tf.cc

```diff
@@ -660,10 +661,32 @@ Type ToLegalElementType(Type type) {
       .Default([&type](Type) { return type; });
 }
 
+bool isQint32ConstantZeroTensor(Value v) {
+  auto constOp = dyn_cast_or_null<TF::ConstOp>(v.getDefiningOp());
+  if (!constOp) return false;
+
+  tensorflow::Tensor out;
+  if (tensorflow::ConvertToTensor(constOp.value(), &out) !=
+      tensorflow::Status::OK())
+    return false;
+
+  int64_t numElems = out.NumElements();
+  auto flat = out.bit_casted_shaped<int32_t, 1>({numElems});
+  for (int64_t i = 0; i < numElems; ++i) {
+    if (flat(i) != 0) return false;
+  }
+  return true;
+}
+
 bool isConstantZeroTensor(Value v) {
   auto castOp = dyn_cast_or_null<TF::CastOp>(v.getDefiningOp());
   if (castOp) return isConstantZeroTensor(castOp->getOperand(0));
 
+  auto ty = v.getType().dyn_cast<RankedTensorType>();
+  if (ty && ty.getElementType().isa<mlir::TF::Qint32Type>()) {
+    return isQint32ConstantZeroTensor(v);
```

Comment:
Is the num_elements = 1 or num_elements > 1 allowded if it's qint32 type?

Reply:
I add a check to protect empty tensor. `num_elements` > 1 is ok.